### PR TITLE
more RTL

### PR DESCRIPTION
**Pull request by karkir0003**
More RTL. Testing `Feedback.tsx` and `Footer.tsx`

I also wrote some basic unit tests for frontend files in `src/components/general`

### CODE REVIEW

**Comment by karkir0003** [comment]
@farisdurrani for 🍒 on top, I was able to use `Codium.AI` extension to generate unit tests for `DButton.tsx` and had to do very very little modification to get the unit test to work 💡 🧠

File: frontend/playground-frontend/src/components/Footer/Footer.tsx [hunk 1]
```diff
@@ -5,29 +5,29 @@ import React from "react";
...
+    <div id="footer" data-testid="footer">
       <IconContext.Provider
         value={{
           color: COLORS.dark_blue,
           size: "2.0rem",
         }}
       >
-        <div className="footer-element" id="footer-socials">
-          <a className="foot-element" href={URLs.linkedin}>
+        <div className="footer-element" id="footer-socials" data-testid="footer-socials">
```
Comment (karkir0003) [inline frontend/playground-frontend/src/components/Footer/Footer.tsx:15]: `data-testid` is a helpful field you can pass into a div to make it easily queryable when it comes to unit testing on the frontend. This field should be unique to the particular div within the confines of the given file

File: frontend/playground-frontend/src/components/general/DButton.test.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    const button = getByRole("button");
+    expect(button).toBeInTheDocument();
+    expect(button).toHaveClass("btn", "btn-primary");
+    expect(button).not.toBeDisabled();
+    expect(button).toHaveTextContent("");
+  });
+
+  // Tests rendering a button with custom properties. tags: [happy path]
+  it("test_custom_button", () => {
+    const onClick = jest.fn();
```
Comment (karkir0003) [inline frontend/playground-frontend/src/components/general/DButton.test.tsx:20]: jest.fn() is a way to mock a function in RTL

File: frontend/playground-frontend/src/components/helper_functions/TalkWithBackend.test.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      displayName: "myname",
+    },
+  },
+}));
+
+describe("Test uploadToBackend function", () => {
+    it("should call axios.post with correct arguments", async () => {
+        const data = { key1: "value1", key2: "value2" };
+        const token = "mytoken";
+        (auth.currentUser?.getIdToken as jest.Mock).mockResolvedValueOnce(token);
```
Comment (karkir0003) [inline frontend/playground-frontend/src/components/helper_functions/TalkWithBackend.test.tsx:26]: sometimes when it comes to firebase, we need to cast functions as a Mock so that we can control what the "resolved value" is or the implementation. weird edge case when it comes to unit testing JS/TS code with `jest`

**Comment by karkir0003** [comment]
added a few comments in certain lines of code in the test files to clarify what the line does.

Also, I started up `TalkWithBackend.test.tsx` and have a few basic unit tests. Feel free to add more as you see fit. Recommended to use Codium.AI to help generate more test cases and correct the code it generates accordingly till you get the test to pass

File: frontend/playground-frontend/src/components/Feedback/Feedback.tsx [hunk 4]
```diff
@@ -2,7 +2,10 @@
 import React, { useState } from "react";
 import type { ReactNode } from "react";
 
-import { EmailInput, TitleText, Spacer } from "../index";
+import EmailInput from "../Home/EmailInput";
+import TitleText from "../general/TitleText";
+import Spacer from "../general/Spacer";
+//import { EmailInput, TitleText, Spacer } from "../index";
```
Comment (karkir0003) [inline frontend/playground-frontend/src/components/Feedback/Feedback.tsx:8]: we actually don't need this line here b/c it was causing the Feedback.test.tsx unit tests to fail

**Comment by farisdurrani** [comment]
Very good, can you please find out why the build checks are failing when you can? Thanks!

**Comment by karkir0003** [comment]
> Very good, can you please find out why the build checks are failing when you can? Thanks!

im debugging it. its due to import in `TalkWithBackend.tsx`

**Comment by karkir0003** [comment]
@farisdurrani build check is failing because I'm having trouble passing in the props for FRQuestion. Could you checkout this branch and go to `FRQuestion.test.tsx` and try to modify the `mockProps`, `mockQuestionObject` so that the `FRQuestion` component renders properly. We need to get this resolved so that the unit tests pass + build check passes. 

What I can do is stash the commit for FRQuestion and move it to another branch for now

**Review by farisdurrani — APPROVED** [review]

File: frontend/playground-frontend/package.json [hunk 5]
```diff
@@ -15,15 +16,15 @@
...
+    "@types/jest": "^29.5.0",
     "@types/node": "^18.15.3",
     "@types/react": "^18.0.28",
     "@types/react-dom": "^18.0.11",
     "@types/react-google-recaptcha": "^2.1.5",
     "@types/redux-mock-store": "^1.0.3",
     "@types/styled-components": "^5.1.26",
     "@uiw/react-codemirror": "^4.12.3",
-    "axios": "^0.27.2",
+    "axios": "0.27.2",
```
Comment (farisdurrani) [inline frontend/playground-frontend/package.json:27]: Any reason for fixing the version of axios?
